### ISSUE TITLE
Crash 'session.isExpired is not a function'

### ISSUE THREAD

**Issue by Tiim** [SerayaEryn/fastify-session#82]
I'm getting the following crash since updating fastify-session from version 3.0.0 to version 3.1.0:

```
/code/node_modules/bluebird/js/release/async.js:49
        fn = function () { throw arg; };
                           ^

TypeError: session.isExpired is not a function
    at /code/node_modules/fastify-session/lib/fastifySession.js:61:23
    at tryCatcher (/code/node_modules/bluebird/js/release/util.js:16:23)
    at Promise.successAdapter [as _fulfillmentHandler0] (/code/node_modules/bluebird/js/release/nodeify.js:23:30)
    at Promise._settlePromise (/code/node_modules/bluebird/js/release/promise.js:601:21)
    at Promise._settlePromise0 (/code/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/code/node_modules/bluebird/js/release/promise.js:729:18)
    at _drainQueueStep (/code/node_modules/bluebird/js/release/async.js:93:12)
    at _drainQueue (/code/node_modules/bluebird/js/release/async.js:86:9)
    at Async._drainQueues (/code/node_modules/bluebird/js/release/async.js:102:5)
    at Immediate.Async.drainQueues (/code/node_modules/bluebird/js/release/async.js:15:14)
    at processImmediate (internal/timers.js:439:21)
```

For a session store I use the following package:
[connect-session-knex](https://github.com/llambda/connect-session-knex)

**Comment by aiibe** [comment]
I second that. I followed this [example](https://github.com/llambda/connect-session-knex/blob/master/example-postgres.js) for Express.
Also using fastify-session v3.1.0 with connect-session-knex v1.5.0
```javascript
const fastify = require('fastify')({ logger: true, ignoreTrailingSlash: true })
const db = require('./db') // knex instance
const cookie = require('fastify-cookie')
const session = require('fastify-session')
const sessionStore = require('connect-session-knex')(session)

// Session
fastify.register(cookie)
fastify.register(session, {
    secret: 'my-secret-string',
    store: new sessionStore({ knex: db }),
    cookie: { maxAge: 10000 }, // https via ngrok
})
```

**Comment by aiibe** [comment]
@Tiim Try this. I got it working by adding `saveUninitialized: false` in the session config object like so...

```javascript
fastify.register(session, {
    secret: 'my-secret-string',
    saveUninitialized: false, // add this line
    store: new sessionStore({ knex: db }),
    cookie: { maxAge: 10000 }
})
```

**Comment by SerayaEryn** [comment]
A fix was released with v3.1.1
